File: ChatGPT/Direct_Attempt/clojure/output/output_22a.clj
;; ChatGPT-generated solution will go here.
(ns output_22a
  (:require [clojure.string :as str]))

(defn mix [secret val]
  (bit-xor secret val))

(defn prune [secret]
  (mod secret 16777216))

(defn next-secret [secret]
  (-> secret
      (mix (* secret 64))
      prune
      (#(mix % (quot % 32)))
      prune
      (#(mix % (* % 2048)))
      prune))

(defn nth-secret [start n]
  (nth (iterate next-secret start) n))

(defn part [lines]
  (->> lines
       (map #(Long/parseLong %))
       (map #(nth-secret % 2000))
       (reduce +)))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))

File: ChatGPT/Direct_Attempt/python/output/output_22a.py
# ChatGPT-generated solution will go here.
def next_secret(secret):
    secret ^= secret * 64
    secret %= 16777216
    secret ^= secret // 32
    secret %= 16777216
    secret ^= secret * 2048
    secret %= 16777216
    return secret

def main():
    with open("input.txt") as f:
        secrets = [int(line.strip()) for line in f if line.strip()]

    total = 0
    for secret in secrets:
        for _ in range(2000):
            secret = next_secret(secret)
        total += secret

    print(total)

if __name__ == "__main__":
    main()
